Added device condition to signon policy API in the dev doc (#2157)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/policy/index.md

}
```

#### Device Condition object

<ApiLifecycle access="ie" /><br>

> 

Reviewer: Please add the note here, as per out guidance:

```
> **Note:** This object is only available for Okta Identity Engine.
```

Author: Thanks will do

Specifies the device condition to match on

| Parameter | Description              | Data Type | Required |
| ---       | ---                      | ---       | ---      |
| registered   | If the device is registered | Boolean     | No      |
| managed   | If the device is managed | Boolean     | No      |

> > **Note:** When managed is set, registered is required, and the value must be true.

Reviewer: ```suggestion
> **Note:** When `managed` is `TRUE`, `registered` is required and must be `TRUE`.
```

Author: what I mean is that if you specify `manage` attribute, not matter the value is `TRUE` or `FALSE`, the `register` must be `TRUE`. The following is not allowed:
{
"managed": true
}
or
{
"managed": false
}

Reviewer: Updated my suggestion. How's that?

Author: sounds good. thanks!

}
```

#### Device Condition object

<ApiLifecycle access="ie" /><br>

> **Note:** This feature is only available as a part of the Okta Identity Engine. Please [contact support](mailto:dev-inquiries@okta.com) for further information.

Specifies the device condition to match on

| Parameter | Description              | Data Type | Required |
| ---       | ---                      | ---       | ---      |
| registered   | If the device is registered | Boolean     | No      |

> | managed   | If the device is managed | Boolean     | No      |

Reviewer 2: Should we elaborate a bit more on what `registered` and `managed` means here?  Registered => Registered with okta or (registered with OV). Managed => device is managed by an MDM and it should have a link to the Device Management section here

}
```

#### Device Condition object

<ApiLifecycle access="ie" /><br>

> **Note:** This feature is only available as a part of the Okta Identity Engine. Please [contact support](mailto:dev-inquiries@okta.com) for further information.

Specifies the device condition to match on

| Parameter | Description              | Data Type | Required |
| ---       | ---                      | ---       | ---      |
| registered   | If the device is registered. A device is registered when the user enrolls with Okta Verify installed on the device | Boolean     | No      |
| managed   | If the device is managed. A device is managed when the device is managed by the device management system| Boolean     | No      |

> **Note:** When `managed` is passed, `registered` must also be included and must be set to `true`.

> For details about integration with device management system, see

Reviewer 3: ```suggestion
For details on integration with a device management system, see
```

 - [Configure Device Trust on Identity Engine for desktop devices](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/config-desktop.htm)
 - [Configure Device Trust on Identity Engine for mobile devices](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/devices/config-mobile.htm)
#### Device Condition object example

```json
"device": {
  "registered": true,
  "managed": true
}
```

> #### Risk Score Condition Object

Reviewer 3: ```suggestion
#### Risk Score Condition object
```

> **Note:** The indicated objects and properties below are only available as a part of the Okta Identity Engine. Please contact support for further information.

| Property                                                       | Data Type   | Description                                                                                 | Supported Values                | Required | Default
| ---                                                            | ---         | ---                                                                                         | ---                             | ---      | ---
| `access`                                                       | String      | Indicates if the action is permitted                                                        | `ALLOW`, `DENY`                 | No       | `DENY`
| `requirement` <ApiLifecycle access="ie" />                     | Object      | JSON object containing Authenticator methods required to be verified if `access` is `ALLOW`. If access is `ALLOW` and `requirement` is not specified, `recovery.factors` from parent policy object is used to determine recovery factors.                             | No       |

> | `requirement.primary.methods` <ApiLifecycle access="ie" />     | Array       | Authenticator methods that can be used by end user to initiate password recovery            | `EMAIL`, `SMS`, `VOICE`, `PUSH` | Yes |

Reviewer 3: ```suggestion
| `requirement.primary.methods` <ApiLifecycle access="ie" />     | Array       | Authenticator methods that can be used by the End User to initiate a password recovery            | `EMAIL`, `SMS`, `VOICE`, `PUSH` | Yes |
```

> | `requirement.stepUp.required` <ApiLifecycle access="ie" />     | Boolean     | Indicates if any step up verification is required to recover password following primary methods verification | `true`, `false` | Yes |

Reviewer 3: ```suggestion
| `requirement.stepUp.required` <ApiLifecycle access="ie" />     | Boolean     | Indicates if any step-up verification is required to recover a password that follows a primary methods verification | `true`, `false` | Yes |
```
